csharp/src/Google.Protobuf/ByteString.cs

```diff
@@ -411,23 +471,55 @@ public bool Equals(ByteString other)
         /// </summary>
         internal void WriteRawBytesTo(CodedOutputStream outputStream)
         {
+#if GOOGLE_PROTOBUF_SUPPORT_SYSTEM_MEMORY
+            if (MemoryMarshal.TryGetArray(bytes, out ArraySegment<byte> segment))

```

Comment:
To avoid possible redundant allocations I suggest adding span-based `WriteRawBytes` to `CodedOutputStream` (internally `ref Span<byte>` is being used).

Reply:
This method is internal and nothing used it. I've remove WriteRawBytesTo entirely.